crates/binstalk-downloader/src/download.rs

```diff
@@ -276,5 +278,35 @@ mod test {
         assert!(!extracted_files.has_file(&dir.join("asdfcqwe")));
 
         assert!(extracted_files.has_file(&dir.join("completions/zsh")));
+
+        // sccache, tgz and zip
+        let sccache_config = [
+            ("https://github.com/mozilla/sccache/releases/download/v0.3.3/sccache-v0.3.3-x86_64-pc-windows-msvc.tar.gz", PkgFmt::Tgz),
+            ("https://github.com/mozilla/sccache/releases/download/v0.3.3/sccache-v0.3.3-x86_64-pc-windows-msvc.zip", PkgFmt::Zip),
+        ];
+
+        for (sccache_url, fmt) in sccache_config {
+            let extracted_files = Download::new(client.clone(), Url::parse(sccache_url).unwrap())
+                .and_extract(fmt, tempdir().unwrap())
+                .await
+                .unwrap();
+
+            let dir = Path::new("sccache-v0.3.3-x86_64-pc-windows-msvc");
+
+            assert_eq!(
+                extracted_files.get_dir(Path::new(".")).unwrap(),
+                &HashSet::from([dir.as_os_str().into()])
+            );
+
+            assert_eq!(
+                extracted_files.get_dir(dir).unwrap(),
+                &HashSet::from_iter(
+                    ["README.md", "LICENSE", "sccache.exe"]
+                        .iter()
+                        .map(OsStr::new)
+                        .map(Box::<OsStr>::from)
+                ),
+            );
+        }
```

Comment:
Not blocking, but feeling uncomfortable about having inline tests that hit the network. These would better live in tests/ behind a feature gate so anyone else running these doesn't get spurious errors because they happen to have network restricted or unavailable.

Reply:
Currently, the tests access the field `ExtractedFiles.0` which is `pub(super)`, so I can't just move it to another file yet.
Would have to either remove that tests or add a new API in a separate PR.